core/app/models/spree/taxon.rb

```diff
@@ -59,8 +59,8 @@ def seo_title
     # name and its parents permalink (if present.)
     def set_permalink
       permalink_tail = permalink.split('/').last if permalink.present?
-      permalink_tail ||= name.to_url
-      self.permalink_part = permalink_tail
+      permalink_tail ||= Spree::Config.string_inflection_class.parameterize(name)
+      self.permalink_part = Spree::Config.string_inflection_class.dasherize(permalink_tail)
```

Comment:
It's not clear to me why we are using two different methods here. Previously we only used one.

Reply:
stringex would "dasherize" everything as part of its `to_url` method. `ActiveSupport::Inflector` doesn't do that, so some [specs related to this](https://github.com/solidusio/solidus/blob/master/core/spec/models/spree/taxon_spec.rb#L71) would fail (`t2_child` is expected to be convereted to `t2-child`, which `ActiveSupport#parameterize` alone does not do).

Reply:
I'd say delete the specs... but thats just me.

Reply:
We can either create a new class which calls `parameterize` then `dasherize`, or we can just use `parameterize`.

Since we're already changing behaviour here I think it is fine to just use `parameterize` and update the specs to pass.